### DISCUSSION TITLE
Downloading debug builds (artifacts) from within GitHub app

### DISCUSSION THREAD

**Discussion by keunes** [community/community#28572] [Mobile] [answered]
Previously the project I'm involved in (AntennaPod app for Android) used CircleCI for unit tests, but they switched to GitHub CI. As an active, non-code contributor this change means unfortunately that I no longer can get the debug builds directly from my phone, to download the apk and test something out.
Three reasons that make this kind of a pain:
* Artifact downloads are not possible from within the app. When tapping on the debug build test, the GitHub website opens instead.
* Logs and debug builds aren't accessible without log-in. As I use Firefox Focus as standard browser, I'd have to log in every single time.
* The single asset from this action/test is an APK file. Nonetheless this single file is wrapped in a ZIP that I'd have to unpack before being able to do anything with it.

My workflow used to be (with CircleCI): go to notifications in GH app, open issue, open debug build page, download apk, install apk, play around, give feedback in the GH app.
Now it is: go to notifications in GH app, open issue, open debug build page, log in, enter 2FA code, find the place with the action artifact, download zip file, unpack the zip, and continue with testing and reporting as mentioned above.

I saw already that the logs & artifacts cannot be made public for security reasons. But if I could download the artifact directly from within the GH app (where I'm authenticated already) and single artifacts weren't zipped, I'd be fine.

Any tips or workarounds would be welcome also :)

**Comment by alcere** [comment, marked as answer]
Hi @keunes, unfortunately there isn't any easy way to do that different from the flow you have described, being said that, the team is currently working on the GitHub Actions integration fopr mobile, so stay tuned for changes! Thanks!

**Reply by keunes** [reply]
Many thanks @alcere. I hope that the described use-case (download single artifact) will be covered in that work :)

**Reply by archon810** [reply]
@alcere is there any sign of artifact downloading coming to Android anytime soon? It's really weird that it shows you "4 Artifacts", it's tappable, but a tap opens some activity and immediately closes it.

**Reply by x9136** [reply]
@alcere You can partially improve the situation if you make a little trick. Make direct links that are accessible from the web version available in the app. As a result, tapping on the artifact will open the link (in the browser) and the download will begin immediately (if you have logged into your account at least once before).

**Comment by 05397477802** [comment]
05397477802

**Comment by codyaintgoing** [comment]
@***Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All

**Comment by codyaintgoing** [comment]
- ### 

> [`_``_`](UFO)# Synchronize with the remote main branch git checkout main git pull origin main # Delete all local branches except for main git branch | grep -ve "main" | xargs git branch -D # List all remote branches except for main branches=$(git branch -r 2> /dev/null | grep -ve "main" | cut -d "/" -f 2) # Synchronize each branch with main and push the result while IFS= read -r branch; do # Fetch and switch to feature branch git fetch origin $branch git checkout $branch # Ensure local environment is free of extra files git clean -xdf # Merge down main git merge --no-ff main # Break out if merge failed if [ $? -ne 0 ]; then break fi # Push update git push origin $branch done <<< "$branches" # Switch back to main git checkout mainHelloWorld/wwwroot/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.min.js<?xml version="1.0"?>  <rdf:RDF xml:lang="en" xmlns="http://usefulinc.com/ns/doap#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:asfext="http://projects.apache.org/ns/asfext#" xmlns:foaf="http://xmlns.com/foaf/0.1/"> <asfext:pmc rdf:about="freemarker"> <asfext:name>Apache FreeMarker</asfext:name> <asfext:charter> The mission of Apache FreeMarker is the creation and maintenance of software related to a template engine. </asfext:charter> <foaf:homepage rdf:resource="http://freemarker.apache.org/"/> </asfext:pmc> </rdf:RDF>Commerce Control List (CCL) Resources

**Comment by x9136** [comment]
Hi! I agree that we need to be able to download debug versions from the app. Right now this is only possible if you go to the web version and log in.

**Reply by juliancoronado** [reply]
Yup! I'd love to be able to download the .zip artifact from the mobile app. That way my Android builds can be easily tested on my personal device!

**Comment by Franklyn-R-Silva** [comment]
It's unfortunate that the change from CircleCI to GitHub CI has disrupted your workflow. While some of the issues you're facing might be inherent to the platform's design, there could still be some workarounds or adjustments you could consider. Here are a few tips that might help you navigate this new workflow:

1. **Direct Link to Artifacts**: Although the GitHub website opens when you tap on the debug build test, you might want to check if there's a direct link to the artifact. Sometimes, right-clicking on the "Download" link and selecting "Copy link address" can provide you with a direct URL to the artifact. You could then paste this URL in a browser or a download manager to get the APK without having to go through the GitHub website interface.

2. **Alternative Browser for GitHub**: Since you mentioned that you have to log in every time using Firefox Focus, you could try using a different browser that might offer better support for maintaining your GitHub session. Popular browsers like Chrome, Firefox, or Edge might provide more seamless authentication.

3. **Automation Tools**: If you're comfortable with scripting or automation, you could create a script that logs in to GitHub, navigates to the appropriate page, and downloads the artifact for you. This would require some programming knowledge, but it could significantly streamline your workflow.

4. **Mobile CI/CD Apps**: There are mobile apps available for some CI/CD platforms, including GitHub Actions. These apps might offer a more mobile-friendly interface for accessing logs and artifacts. It's worth checking if there's a GitHub Actions app that could help you access artifacts more easily.

5. **Provide Feedback**: If these changes are significantly affecting your ability to contribute effectively, it might be worth discussing your concerns with the project maintainers. They might not be aware of how these changes are impacting non-code contributors, and they might be open to finding ways to improve the process.

6. **Check for Updates**: As the software ecosystem is constantly evolving, the GitHub app and the CI/CD platform might release updates that address some of the usability issues you're facing. Keep an eye out for app updates that could potentially improve your experience.

Remember that adapting to new workflows can be challenging, but with some creative thinking and exploration, you might find ways to make the process smoother.

**Comment by archon810** [comment]
The latest beta of the Github app version 1.158.0 finally resolves this. Now instead of an endless loop when trying to download artifacts, it opens a web browser and scrolls down to the artifacts.

It's not a complete solution as it relies on opening the browser, but it's much better.

> - Assets in releases now have a more concise UI, allowing for longer asset titles.
> - Fixed crashes in the add to list and repositories screens.
> - Fixed loop accessing workflow run artifacts.
